### PR TITLE
Create reference point for v4.0.0 rc4

### PR DESCRIPTION
**Pull request by shaopeng-gh**
Create reference point for v4.0.0 rc4 without actual release.

* DEP: Update Sarif.Sdk submodule from [235394a to 120fae35](https://github.com/microsoft/sarif-sdk/compare/235394a...120fae35). Full [SARIF SDK Release History](https://github.com/microsoft/sarif-sdk/blob/120fae35/src/ReleaseHistory.md).

### CODE REVIEW

File: src/BinaryParsers/VersionConstants.cs [hunk 1]
```diff
@@ -4,9 +4,9 @@ namespace Microsoft.CodeAnalysis.IL
 {
     public static class VersionConstants
```
Comment (shaopeng-gh) [inline src/BinaryParsers/VersionConstants.cs:5]: >VersionConstants [](http://example.com/codeflow?start=24&length=16)

I found this file, looks like we should add in the Release-Process md file

File: SetCurrentVersion.cmd [hunk 2]
```diff
@@ -1,9 +1,9 @@
-set MAJOR_PREVIOUS=2
+set MAJOR_PREVIOUS=4
 set MINOR_PREVIOUS=0
 set PATCH_PREVIOUS=0
-set PRERELEASE_PREVIOUS=-rc2
+set PRERELEASE_PREVIOUS=-rc3
 
-set MAJOR=2
```
Comment (shaopeng-gh) [inline SetCurrentVersion.cmd:6]: >2 [](http://example.com/codeflow?start=10&length=1)

I think we should have changed to 4 before this PR.

File: src/BinaryParsers/VersionConstants.cs (hunk 1 above)
Comment (suvamM) [reply src/BinaryParsers/VersionConstants.cs:5]: @shaopeng-gh Agree. Please update the release process.

File: src/ReleaseHistory.md [hunk 3]
```diff
@@ -13,7 +13,10 @@
 - UER => eliminate unhandled exceptions in rules
 - UEE => eliminate unhandled exceptions in engine
 
-## **v4.0.0-rc4** UNRELEASED
+## **v4.0.0-rc5** UNRELEASED
+
+## **v4.0.0-rc4**
+* DEP: Update Sarif.Sdk submodule from [235394a to ec93dccd](https://github.com/microsoft/sarif-sdk/compare/235394a...ec93dccd). Full [SARIF SDK Release History](https://github.com/microsoft/sarif-sdk/blob/ec93dccd/src/ReleaseHistory.md).
```
Comment (suvamM) [inline src/ReleaseHistory.md:19]: @shaopeng-gh Note that Mike's PR on threading fixes got merged: https://github.com/microsoft/sarif-sdk/pull/2618

So we should update the `sarif-sdk` submodule to reference the appropriate point in `main`.

File: src/Test.FunctionalTests.BinSkim.Rules/TestMessageLogger.cs [hunk 4]
```diff
@@ -42,7 +42,7 @@ public void AnalyzingTarget(IAnalysisContext context)
         {
         }
 
-        public void Log(ReportingDescriptor rule, Result result)
+        public void Log(ReportingDescriptor rule, Result result, int? extensionIndex = null)
```
Comment (suvamM) [inline src/Test.FunctionalTests.BinSkim.Rules/TestMessageLogger.cs:45]: @shaopeng-gh Again, unclear what this change does.

**Comment by suvamM** [comment]
@shaopeng-gh Please don't leave the PR description blank.

**Review by suvamM — CHANGES_REQUESTED** [review]

File: src/BinSkim.Driver/RuleTelemetryLogger.cs [hunk 5]
```diff
@@ -99,7 +99,7 @@ public void AnalyzingTarget(IAnalysisContext context)
         {
         }
 
-        public void Log(ReportingDescriptor rule, Result result)
+        public void Log(ReportingDescriptor rule, Result result, int? extensionIndex = null)
```
[the remark this answers is not in the document]
Comment (shaopeng-gh) [reply src/BinSkim.Driver/RuleTelemetryLogger.cs:102]: This is due to a breaking change that add a optional parameter:
BRK: Add ToolComponent argument to IAnalysisLogger.Log(ReportingDescriptor, Result) method. [https://github.com/microsoft/sarif-sdk/pull/2611](https://github.com/microsoft/sarif-sdk/pull/2611)

File: SetCurrentVersion.cmd (hunk 2 above)
Comment (shaopeng-gh) [reply SetCurrentVersion.cmd:6]: yes, this was missed in the previous PR, we should have changed it at that time.

File: src/ReleaseHistory.md (hunk 3 above)
Comment (shaopeng-gh) [reply src/ReleaseHistory.md]: Updated to latest main.

File: src/BinaryParsers/VersionConstants.cs (hunk 1 above)
Comment (shaopeng-gh) [reply src/BinaryParsers/VersionConstants.cs:5]: updated

**Comment by shaopeng-gh** [comment]
added.

---
In reply to: [1433423241](https://github.com/microsoft/binskim/pull/811#issuecomment-1433423241) [](http://example.com/codeflow?ancestors=1433423241,1433423241)

**Review by suvamM — APPROVED** [review]

**Pull request merged**